mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BucketNameGenerator.java

```diff
@@ -21,20 +21,45 @@
 
 import java.time.Clock;
 import java.time.ZonedDateTime;
+import java.time.temporal.ChronoField;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import org.apache.james.blob.api.BucketName;
 
 public class BucketNameGenerator {
+    private static final Pattern BUCKET_NAME_PATTERN = Pattern.compile("deleted-messages-([0-9]{4})-([0-9]{2})-(01)");
+    private static final String BUCKET_NAME_GENRATING_FORMAT = "deleted-messages-%d-%02d-01";
+    private static final int FIRST_DAY_OF_MONTH = 1;
+    private static final int ZERO_HOUR = 0;
+    private static final int ZERO_MINUTE = 0;
+    private static final int ZERO_SECOND = 0;
+    private static final int ZERO_NANOSECOND = 0;
+
     private final Clock clock;
 
     public BucketNameGenerator(Clock clock) {
         this.clock = clock;
     }
 
-    public BucketName currentBucket() {
+    BucketName currentBucket() {
         ZonedDateTime now = ZonedDateTime.now(clock);
-        int month = now.getMonthValue();
+        int month = now.getMonth().get(ChronoField.MONTH_OF_YEAR);
         int year = now.getYear();
-        return BucketName.of(String.format("deletedMessages-%d-%02d-01", year, month));
+        return BucketName.of(String.format(BUCKET_NAME_GENRATING_FORMAT, year, month));
+    }
+
+    Optional<ZonedDateTime> bucketEndTime(BucketName bucketName) {
+        return Optional.of(BUCKET_NAME_PATTERN.matcher(bucketName.asString()))
+            .filter(Matcher::matches)
+            .map(matcher -> {
+                int year = Integer.parseInt(matcher.group(1));
+                int month = Integer.parseInt(matcher.group(2));
+                ZonedDateTime firstDayOfMonth = ZonedDateTime.of(year, month, FIRST_DAY_OF_MONTH, ZERO_HOUR, ZERO_MINUTE,
+                    ZERO_SECOND, ZERO_NANOSECOND, clock.getZone());
+                ZonedDateTime lastInstantOfMonth = firstDayOfMonth.plusMonths(1).minusNanos(1);
```

Comment:
I'd rather return an exact non-inclusive upper bound rather than that kind of hack